File: SIN BDD/SOAP_JAVA_CONUNI_GR06/07. CLIMOVIL/cliente_movil_soap_java_conuni_gr06/app/src/main/java/ec/edu/monster/modelo/ServicioMasa.java
package ec.edu.monster.modelo;

import java.util.LinkedHashMap;
import java.util.Map;

public class ServicioMasa {

    private final ClienteSoap clienteSoap = new ClienteSoap();

    public double kilogramosALibras(double kilogramos) throws Exception {
        return invocarUnario("kilogramosALibras", "kilogramos", kilogramos);
    }

    public double gramosAOnzas(double gramos) throws Exception {
        return invocarUnario("gramosAOnzas", "gramos", gramos);
    }

    public double toneladasAKilogramos(double toneladas) throws Exception {
        return invocarUnario("toneladasAKilogramos", "toneladas", toneladas);
    }

    public double librasAOnzas(double libras) throws Exception {
        return invocarUnario("librasAOnzas", "libras", libras);
    }

    public double miligramosAGramos(double miligramos) throws Exception {
        return invocarUnario("miligramosAGramos", "miligramos", miligramos);
    }

    private double invocarUnario(String operacion, String nombreParametro, double valor) throws Exception {
        Map<String, String> parametros = new LinkedHashMap<>();
        parametros.put(nombreParametro, String.valueOf(valor));
        String respuesta = clienteSoap.invocar(operacion, parametros);
        return Double.parseDouble(respuesta);
    }
}

File: SIN BDD/SOAP_DOTNET_CONUNI_GR06/04. CLICONSOLA/cliente_consola_soap_dotnet_conuni_gr06/src/Modelo/ServicioMasa.cs
namespace Ec.Edu.Monster.Modelo;

public class ServicioMasa
{
    private readonly ClienteSoap cliente = new();

    public double KilogramosALibras(double kilogramos) => cliente.Ejecutar(servicio => servicio.KilogramosALibras(kilogramos));

    public double GramosAOnzas(double gramos) => cliente.Ejecutar(servicio => servicio.GramosAOnzas(gramos));

    public double ToneladasAKilogramos(double toneladas) => cliente.Ejecutar(servicio => servicio.ToneladasAKilogramos(toneladas));

    public double LibrasAOnzas(double libras) => cliente.Ejecutar(servicio => servicio.LibrasAOnzas(libras));

    public double MiligramosAGramos(double miligramos) => cliente.Ejecutar(servicio => servicio.MiligramosAGramos(miligramos));
}
